base-template.sh

```diff
@@ -774,61 +787,140 @@ show_prompt() {
+    fi
+
+    if [ -d "$INSTALL_DIR/release" ]; then
+        if ! rm -rf "$INSTALL_DIR/release" >/dev/null 2>&1; then
+            echo "! Failed to remove an existing githook release repository" >&2
             return 1
         fi
     fi
 
-    # shellcheck disable=SC2181
+    echo "Installing a clone of \`$GITHOOKS_CLONE_URL\` to \`$INSTALL_DIR/release\` ..."
+
+    CLONE_OUTPUT=$(git clone \
+        -c core.hooksPath=/dev/null \
+        --depth 1 \
+        --single-branch \
+        --branch "$GITHOOKS_CLONE_BRANCH" \
+        "$GITHOOKS_CLONE_URL" "$INSTALL_DIR/release" 2>&1)
+
+    #shellcheck disable=2181
     if [ $? -ne 0 ]; then
-        echo "! Cannot download file \`$DOWNLOAD_FILE\` - command failed" >&2
+        echo "! Cloning of repository \`$GITHOOKS_CLONE_URL\`"
+        echo "! to install directory \`$INSTALL_DIR/release\` failed with output: " >&2
+        echo "$CLONE_OUTPUT" >&2
         return 1
     fi
+
+    git config --global githooks.autoupdate.releaseCloneUrl "$GITHOOKS_CLONE_URL"
+    git config --global githooks.autoupdate.releaseCloneBranch "$GITHOOKS_CLONE_BRANCH"
+
     return 0
 }
 
-#####################################################
-# Loads the contents of the latest install
-#   script into a variable.
+############################################################
+# Checks whether the given directory
+#   is a Git repository (bare included) or not.
 #
-# Sets the ${INSTALL_SCRIPT} variable
+# Returns:
+#   1 if failed, 0 otherwise
+############################################################
+is_git_repo() {
+    git -C "$1" rev-parse >/dev/null 2>&1 || return 1
+}
+
+#####################################################
+# Updates the release clone in the install folder.
 #
 # Returns:
-#   1 if failed the load the script, 0 otherwise
+#   1 if failed, 0 otherwise
 #####################################################
-fetch_latest_update_script() {
-    echo "^ Checking for updates ..."
+update_release_clone() {
 
-    INSTALL_SCRIPT="$(mktemp)"
-    if ! download_file "install.sh" "$INSTALL_SCRIPT"; then
-        echo "! Failed to check for updates" >&2
-        return 1
+    CLONE_DIR="$INSTALL_DIR/release"
+    PULL_ONLY="true"
+
+    # We do a fresh clone if:
+    # - we have an existing clone AND
+    #   - url or branch of the existing clone does not match the settings OR
+    #   - the existing clone has modifications (should not be the case)
+    # - we dont have an existing clone
+
+    GITHOOKS_CLONE_URL=$(git config --global githooks.autoupdate.releaseCloneUrl)
+    GITHOOKS_CLONE_BRANCH=$(git config --global githooks.autoupdate.releaseCloneBranch)
+
+    if is_git_repo "$CLONE_DIR"; then
+        URL=$(git -C "$CLONE_DIR" config remote.origin.url)
+        BRANCH=$(git -C "$CLONE_DIR" symbolic-ref -q --short HEAD)
+
+        if [ "$URL" != "$GITHOOKS_CLONE_URL" ] ||
+            [ "$BRANCH" != "$GITHOOKS_CLONE_BRANCH" ] ||
+            git -C "$CLONE_DIR" diff --quiet; then
```

Comment:
This means we're going to `rm -rf` and clone over any local changes, right?
I wonder if we could try a clean merge and fail the installation if that doesn't work instead?
We could say, you're local install clone is dirty, either update manually or revert your local changes.
What do you think? It might be too much of a hassle, just wanted to throw it out there as a suggestion.

Reply:
We could do that, might be less intrusive :-) :+1: Will test that.

Reply:
Hm. I think we should not add another indirect complication over a merge of the configured release branch to a possible changed (by the user) release branch. I think we just pull the configured branch, thats it :-). 
However I will introduce a warning, when we are dirty or are on some other branch (the user fiddled with it) we will abort the update with a warning you suggest :-)

Reply:
Ok. I come to the conclusion, that the behavior of completely re-clone if something is fishy is ok IMO. In this way we can also provide a 
```shell
git hooks config [set|print] release-clone-url
git hooks config [set|print] release-clone-branch
```
In the cli.sh. If the user changes this -> on the next update it will gracefully replace the existing clone which is nice.
I will however print a warnin when a pull fails (from whatever reason) and tell the user to error and a suggestion to manually update/fix or delete the release clone repo such that further update will succeed.
Is that ok?

Reply:
Yeah, that sounds good to me. This way they'd know that the local release directory shouldn't change, and hopefully would raise an issue here if they really want that. Otherwise their fork should work as expected if they're using a fork.